src/Nix/Builtins.hs

```diff
@@ -100,7 +101,7 @@ withNixContext mpath action = do
     opts :: Options <- asks (view hasLens)
     let i = value @(NValue m) @(NThunk m) @m $ nvList $
             map (value @(NValue m) @(NThunk m) @m
-                     . flip nvStr mempty . Text.pack) (include opts)
+                     . nvStr . hackyMakeNixStringWithoutContext . Text.pack) (include opts)
```

Comment:
Does the function really need to be named `hackyMakeNixStringWithoutContext`? Is this some indication that the work is yet undone?

Reply:
Yes. We need to ensure we look at all places where we create strings and set the context appropriately. For some of those it will be correct to just make a string without context, of course, but currently *every* string creation in hnix is without context and that's definitely wrong.

Reply:
@shlevy As you mentioned before, there are two more places where we use mconcat (Eval.hs) and mappend (Exec.hs) on NixStrings. Should we just get rid of the semigroup/monoid instance and replace those calls with NixString specific calls?